### ISSUE TITLE
Incorrect Formatting of Function Call With Trailing Comment

### ISSUE THREAD

**Issue by sanssecours** [cheshirekow/cmake_format#56]
# Steps to Reproduce

1. Create a file `test.cmake` with the following content

   ```cmake
   set (answer 
        42 # comment  
       )
   ```

3. Run `cmake-format`

   ```sh
   cmake-format test.cmake
   ```

# Expected Result

The command `cmake-format` prints a nicely formatted version of the given CMake code.

# Actual Result

The command produces the following syntactically incorrect code:

```cmake
set(answer 42 # comment)
```

.

**Comment by cheshirekow** [comment]
I think this is the same as #55. A comment should preclude formatting to a single line. Hopefully can fix them both quickly.

**Comment by cheshirekow** [comment]
Ah, on second thought, I see this is a different issue: The paren can't be packed onto the same line because of the comment.

**Comment by michaelmaguire** [comment]
Thanks for this tool!

This ticket's issue still seems to be a problem for me in 0.4.3 (which from 'releases' ought to contain the fix already):

Input:

```
target_link_libraries(peerdiscovery PUBLIC
    ${COMMON_LIBRARIES}
    # add more library dependencies here
)
```

Output:

```
target_link_libraries(peerdiscovery
                      PUBLIC ${COMMON_LIBRARIES}
                             # add more library dependencies here)
```

which produced CMake errors:

```
[windows_build_and_unittest] CMake Error at peerdiscovery/CMakeLists.txt:52:
[windows_build_and_unittest]   Parse error.  Function missing ending ")".  End of file reached.
```

**Comment by unapiedra** [comment]
I still see this. Here is an example:

Input: 
```
cmake_minimum_required(VERSION 3.13.0)
project(minimal_example)

find_package(foobar REQUIRED
             COMPONENTS some_component
                        # some_other_component # This is a very long comment, and actually the second comment in this row.
          )
```

Output:
```
cmake_minimum_required(VERSION 3.13.0)
project(minimal_example)

find_package(foobar REQUIRED
             COMPONENTS some_component
                        # some_other_component # This is a very long comment,
                        # and actually the second comment in this row.)
```
Notice the closing parens on the last comment line.

**Comment by cheshirekow** [comment]
Added both examples to the tests and I believe this is fixed in `0.4.4`. Thanks @siedentop for tracking it down in the source.

**Comment by siedentop** [comment]
I can confirm that this is fixed now. Thank you, @cheshirekow !
